pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -689,6 +726,126 @@ func (r *defaultKialiReconciler) reconcileKiali(ctx context.Context, kialiCRName
 	return nil
 }
 
+type PrometheusReconciler interface {
+	reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, configuredMembers []string) error
+}
+
+type defaultPrometheusReconciler struct {
+	Client client.Client
+}
+
+func (r *defaultPrometheusReconciler) reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, members []string) error {
+	reqLogger := common.LogFromContext(ctx)
+	reqLogger.Info("Attempting to get Prometheus ConfigMap", "prometheusNamespace", prometheusNamespace, "prometheusCMName", prometheusCMName)
+
+	cm := &corev1.ConfigMap{}
+
+	err := r.Client.Get(ctx, client.ObjectKey{Name: prometheusCMName, Namespace: prometheusNamespace}, cm)
+	if err != nil {
+		if meta.IsNoMatchError(err) || errors.IsNotFound(err) || errors.IsGone(err) {
+			reqLogger.Info("Prometheus ConfigMap '%s/%s' does not exist", prometheusNamespace, prometheusCMName)
+			return nil
+		}
+		return pkgerrors.Wrap(err, "error retrieving Prometheus ConfigMap from mesh")
```

Comment:
Why do you validate this config map? Reconciliation should not fail when a user deletes or modifies the config map which we created. Instead, the operator should self-heal itself during reconciliation. So I would expect that if the config map does not exist or it's content is invalid, the operator recreates it.

Reply:
The `configmap.yml` is part of helm chart, which gets applied [here](https://github.com/maistra/istio-operator/pull/1089/files#diff-b86b0b1932da7be356a0558098c3fe6c57d214953652145e2420b357b315ec01R433). During reconciliation, there is no helm chart rendering. Marko suggested we should not introduce the rendering during reconciliation. If we had helm chart rendering, then we will not need the "long" logic to change on some particular fields in the configmap.